livre/lib/cache/HistogramObject.cpp

```diff
@@ -57,11 +92,9 @@ void binData( const SRC_TYPE* rawData,
                     const size_t index = compCount * i * dataBlockSize.y() * dataBlockSize.z() +
                                          compCount * j * dataBlockSize.z() +
                                          compCount * k + c;
-                    const double data = rawData[ index ];
-                    const size_t binIndex =
-                            std::is_unsigned< SRC_TYPE >::value ?
-                            ( data - minVal ) / range * ( binCount - 1 ):
-                            (( data / range ) + 0.5f ) * ( binCount - 1 );
+
+                    const size_t binIndex = ( rawData[ index ] - histogram.getMin( )) /
+                                            range * ( binCount - 1 );
```

Comment:
This operation is not correct for integral types. Imagine you have a 16 bit data range and you want to map it to a histogram of 1024 bins. The mapping of values is [n, 64 * n) -> n, so 64 goes into bin 1, right? However if you compute 64 / 65535.0 * 1023.0 the result is 0.999... instead of 1. 

A correct solution is ( rawData[ index ] - histogram.getMin( )) / (range / binCount);